test/integration/scheduler_perf/util.go

```diff
@@ -53,15 +56,27 @@ const (
 
 var dataItemsDir = flag.String("data-items-dir", "", "destination directory for storing generated data items for perf dashboard")
 
+func newDefaultComponentConfig() (*config.KubeSchedulerConfiguration, error) {
```

Comment:
The logic here first instantiates an up-to-date versioned CC, and calls scheme.Default() to apply default values, and finally convert it back to an unversioned CC. @adtac @alculquicondor could you confirm it's the best practice so far to instantiate a default unversioned CC?

Reply:
This is what scheduler does today,  pls let me know whether you want this this public and import from there.
https://github.com/kubernetes/kubernetes/blob/fcee7a01050652e54d2819b1942533d96e40f455/cmd/kube-scheduler/app/options/options.go#L137-L147

Reply:
Yes, this is how I'd do it too. We only have defaulting for the versioned types, so there's no way to directly apply defaults to the internal type.

Reply:
`scheduler.New` already knows how to handle defaults... why do it again?

Reply:
there's more to using the v1beta1 defaults than what `defaultSchedulerOptions` offers. for example we have default QPS limits set in the v1beta1 defaults.go (which we don't use here right now, but will in the future (https://github.com/kubernetes/kubernetes/pull/99638#discussion_r593025201)), but scheduler_perf should respect that in the future. the `scheduler.New` defaults won't give you that IIUC.

Reply:
> there's more to using the v1beta1 defaults than what defaultSchedulerOptions offers

+1, hope this answer your question?  @alculquicondor

Reply:
The second reason why I add this in the first place is because when the `config` is `nil`,  `schedulerOptions` will be overridden by `nil` even though `scheduler.New` has defined the default options, for example, `cfg.Profiles` carried over to here could be `nil` when `config` is `nil.
We should make sure those options as least have a default value set in this case.

```
	options := defaultSchedulerOptions
	for _, opt := range opts {
		opt(&options)
	}
```

How the config handled here is a little different with scheduler to simplify the implementation.

Reply:
> Then just use Decode

well, both of the methods can be used to set the default options, but the `Decode` here will decode into `runtime.Object`

```
func (c *codec) Decode(data []byte, defaultGVK *schema.GroupVersionKind, into runtime.Object) (runtime.Object, *schema.GroupVersionKind, error)
```

There is no conversion provided to convert `runtime.Object` into internal `KubeSchedulerConfiguration`, but we still need to create a un-versioned scheduler config.

Besides, I not sure why you suggest to use `Decode`, am I miss something?

Reply:
`runtime.Object` is just an interface for any API type, including `KubeSchedulerConfiguration`. Again, see https://github.com/kubernetes/kubernetes/blob/fcee7a01050652e54d2819b1942533d96e40f455/pkg/scheduler/framework/runtime/framework.go#L411-L412 as an example.